### PR TITLE
feat: history

### PR DESCRIPTION
**Pull request by tusharkandpal**
TO CHECK:

- added `historySlice.js`
- created `History.jsx` page to show  user history
- `getHistory.js`, `removeFromHistory.js`, `removeAllHistory.js` & `postToHistory.js` services for api calls.

**Note:**  refactored other components accordingly.

### CODE REVIEW

File: src/component/card/VideoCard.jsx [hunk 1]
```diff
@@ -28,19 +30,30 @@ export const VideoCard = (video) => {
...
 
   return (
     <article className="md:w-[18rem] lg:w-[20.5rem] hover:shadow-md hover:shadow-violet-700/50">
-      <Link to={`/video/${_id}`} className="relative">
+      <Link
+        to={`/video/${_id}`}
+        className="relative"
+        onClick={() =>
+          isLoggedIn && !isVideoInHistory && dispatch(postToHistory(video))
+        }
```
Comment (dev-enforced) [inline src/component/card/VideoCard.jsx:55]: Good logic of using History dispatch action only at the time when the user is getting logged in!!

File: src/services/historyServices/getHistory.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  } catch (error) {
+    dispatch(
+      addToast({
+        type: "WARNING",
+        desc: "Error getting History.",
+      })
+    );
+    return rejectWithValue("Error getting History.");
+  }
+};
```
Comment (dev-enforced) [inline src/services/historyServices/getHistory.js:19]: Why are we adding toast related data in this service? Shouldn't it be kept separate from the data fetch API call?

File: src/features/historySlice.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    return await addToHistory(video, rejectWithValue);
+  }
+);
+
+export const deleteFromHistory = createAsyncThunk(
+  "likes/deleteFromHistory",
+  async (videoId, { rejectWithValue, dispatch }) => {
+    return await removeFromHistory(videoId, rejectWithValue, dispatch);
+  }
+);
```
Comment (dev-enforced) [inline src/features/historySlice.js:34]: This format of writing Async Thunks is very new for me!! Definitely one of the learning points from this pull request.

**Review by dev-enforced — APPROVED** [review]
The feature is working correctly @tusharkandpal as mentioned in the pull request review. Good work on developing this feature!!

File: src/services/historyServices/getHistory.js (hunk 2 above)
Comment (tusharkandpal) [reply src/services/historyServices/getHistory.js:19]: From what I think you're asking, I want to show the toast if the error occurs. Which won't necessarily be the case usually.

File: src/features/historySlice.js (hunk 3 above)
Comment (tusharkandpal) [reply src/features/historySlice.js:34]: Same here ...new & learning from others

**Pull request merged**